### DISCUSSION TITLE
sea-orm-cli --model-extra-derives

### DISCUSSION THREAD

**Discussion by zchira** [SeaQL/sea-orm#2033] [Q&A]
I am trying to use sea-orm with poem. All poem entities should derive poem_openapi_derive::Object trait.
When I call sea-orm-cli with option:
`--model-extra-derives=poem_openapi_derive::Object`

I am getting generated entities like this:
```
#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq, poem_openapi_derive :: Object)]   

```

with space before and after `::` 

Not expected behavior I guess?
Any workaround/advise?

**Comment by tyt2y3** [comment]
`rustfmt` does this, strangely, which is different from `cargo fmt`. It still compiles. I am not sure whether there is an option to change that.

**Reply by zchira** [reply]
Yes, you are right! It compiles. 
The other problem is still there:
I have enum entities. Example:
```
#[derive(Debug, Clone, PartialEq, Eq, EnumIter, DeriveActiveEnum)]
#[sea_orm(rs_type = "String", db_type = "Enum", enum_name = "role")]
pub enum Role {
    #[sea_orm(string_value = "Admin")]
    Admin,
    #[sea_orm(string_value = "User")]
    User
}
```

Since this is not struct/model, `poem_openapi_derive::Object` is not added in `derive` section.
But, this causes compile errors because for enum types in `poem` we need to derive  `poem_openapi_derive::Enum`.

Adding that manually fixes the problem:
```
#[derive(Debug, Clone, PartialEq, Eq, EnumIter, DeriveActiveEnum, poem_openapi_derive::Enum)]
#[sea_orm(rs_type = "String", db_type = "Enum", enum_name = "role")]
```

Any solution/workaround for that?
